src/bio/bio_buffer.c

```diff
@@ -516,7 +515,7 @@ dma_map_one(struct bio_desc *biod, struct bio_iov *biov,
 			dma_free_chunk(chk);
 			return rc;
 		}
-		biov->bi_buf = chk->bdc_ptr + pg_off;
+		bio_iov_set_raw_buf(biov, chk->bdc_ptr + pg_off);
```

Comment:
If you set bi_data_len & bi_addr.ba_off as requested, actually it's not necessary to convert bi_buf to raw, you can keep bi_buf as requested as well, but do the mapping based on raw len & off, then checksum code may need to access raw buf by 'bi_buf - prefix_len' & 'bi_data_len + prefix_len + suffix_len'.

I don't quite see why you want bi_buf as raw, but bi_data_len & bi_addr.ba_off as requested, that looks quite confusing to me. (I actually prefer to set bi_buf, bi_data_len & bi_addr.ba_off as raw.)

Reply:
I changed bi_data_len and bi_addr.ba_off to be raw so it's consistent now.